lang/ruby/lib/avro/io.rb

```diff
@@ -423,11 +423,13 @@ def read_default_value(field_schema, default_value)
             field_val = read_default_value(field.type, json_val)
             read_record[field.name] = field_val
           end
-          return read_record
+          read_record
         else
           fail_msg = "Unknown type: #{field_schema.type}"
           raise AvroError, fail_msg
         end
+
+        field_schema.logical_type ? field_schema.type_adapter.decode(datum) : datum
```

Comment:
I see, because of the Identity
https://github.com/apache/avro/blob/01ae3ca6549cc740396140defe891691e0e3a532/lang/ruby/lib/avro/logical_types.rb#L244-L252